ovos_PHAL_plugin_homeassistant/logic/connector.py

```diff
@@ -308,11 +308,13 @@ def send_assist_command(self, command, arguments={}):
 
 
 class HomeAssistantWSConnector(HomeAssistantConnector):
-    def __init__(self, host, api_key, assist_only=True):
+    def __init__(self, host, api_key, assist_only=True, max_ws_message_size=5242880):
         super().__init__(host, api_key, assist_only)
         if self.host.startswith("http"):
-            self.host.replace("http", "ws", 1)
-        self._connection = HomeAssistantClient(self.host, self.api_key)
+            self.host = self.host.replace("http", "ws", 1)
+        if self.host.startswith("https"):
+            self.host = self.host.replace("https", "wss", 1)
+        self._connection = HomeAssistantClient(url=self.host, token=self.api_key, max_ws_message_size=max_ws_message_size)
```

Comment:
This condition functionally never happens, but at least the logic is now correct